pkg/dispatcher/dispatcher.go

```diff
@@ -0,0 +1,204 @@
+package dispatcher
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/buraksezer/consistent"
+	"github.com/cespare/xxhash"
+	mapset "github.com/deckarep/golang-set/v2"
+	"github.com/openshift-online/maestro/pkg/api"
+	"github.com/openshift-online/maestro/pkg/dao"
+	logger "github.com/openshift-online/maestro/pkg/logger"
+)
+
+// Dispatcher interface outlines methods for coordinating resource status updates
+// in the context of multiple active Maestro instances. Each instance subscribes
+// to a shared topic for resource status updates.
+//
+// The dispatcher's role is to ensure that only one instance processes specific
```

Comment:
not sure it is clear. But I think the purpose of the this interface is to manage the relationship between maestro instance and consumer (agent)?